From f3905d3510dfb3851f946f097a9e2ddaa5fb333b Mon Sep 17 00:00:00 2001
From: Igor Freire <igor.auad@gmail.com>
Date: Mon, 13 May 2019 16:00:07 -0300
Subject: [PATCH 7/7] Include gnuradio-blocks on the cmake build

---
 CMakeLists.txt     | 4 ++--
 lib/CMakeLists.txt | 1 +
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f836233..76375dc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -167,10 +167,10 @@ endif(APPLE)
 ########################################################################
 set(GR_REQUIRED_COMPONENTS RUNTIME PMT BLOCKS)
 
-#[[find_package(gnuradio-blocks PATHS ${GR_PREFIX}/lib/cmake/gnuradio/)
+find_package(gnuradio-blocks PATHS ${GR_PREFIX}/lib/cmake/gnuradio/)
 message(STATUS "Found Block Block: ${gnuradio-blocks_FOUND}")
 
-find_package(gnuradio-pmt PATHS ${GR_PREFIX}/lib/cmake/gnuradio/)
+#[[find_package(gnuradio-pmt PATHS ${GR_PREFIX}/lib/cmake/gnuradio/)
 message(STATUS "Found PMT Block: ${gnuradio-pmt_FOUND}")
 
 find_package(gnuradio-runtime PATHS ${GR_PREFIX}/lib/cmake/gnuradio/)
diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 29fd6d0..2171df2 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -51,6 +51,7 @@ endif()
 
 GR_OSMOSDR_APPEND_LIBS(
     ${Boost_LIBRARIES}
+    gnuradio::gnuradio-blocks
     ${GNURADIO_ALL_LIBRARIES}
 )
 
-- 
2.20.1

