From 46bb1ad1a08b44f9c06dd2bf099731aa3bc0e995 Mon Sep 17 00:00:00 2001
From: Dimitri Stolnikov <horiz0n@gmx.net>
Date: Tue, 16 Dec 2014 00:14:35 +0100
Subject: [PATCH 08/48] airspy: increase version string size to 128

---
 lib/airspy/airspy_source_c.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/airspy/airspy_source_c.cc b/lib/airspy/airspy_source_c.cc
index bc33fed..395b812 100644
--- a/lib/airspy/airspy_source_c.cc
+++ b/lib/airspy/airspy_source_c.cc
@@ -100,7 +100,7 @@ airspy_source_c::airspy_source_c (const std::string &args)
   ret = airspy_board_id_read( _dev, &board_id );
   AIRSPY_THROW_ON_ERROR(ret, "Failed to get AirSpy board id")
 
-  char version[40];
+  char version[128];
   memset(version, 0, sizeof(version));
   ret = airspy_version_string_read( _dev, version, sizeof(version));
   AIRSPY_THROW_ON_ERROR(ret, "Failed to read version string")
-- 
2.1.4

