From 3532d60c11529c8aeca58219bb06f2c2acf8e547 Mon Sep 17 00:00:00 2001
From: Josh Blum <josh@joshknows.com>
Date: Wed, 22 Apr 2015 22:17:24 -0700
Subject: [PATCH 018/131] osmo: minor msvc fixes for bladerf_common.h

---
 lib/bladerf/bladerf_common.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/lib/bladerf/bladerf_common.h b/lib/bladerf/bladerf_common.h
index f67600f..d16c02f 100644
--- a/lib/bladerf/bladerf_common.h
+++ b/lib/bladerf/bladerf_common.h
@@ -31,6 +31,7 @@
 #include <boost/assign.hpp>
 #include <boost/format.hpp>
 #include <boost/lexical_cast.hpp>
+#include <boost/weak_ptr.hpp>
 
 #include <gnuradio/thread/thread.h>
 #include <gnuradio/gr_complex.h>
@@ -40,6 +41,11 @@
 #include "osmosdr/ranges.h"
 #include "arg_helpers.h"
 
+#ifdef _MSC_VER
+#include <cstddef>
+typedef ptrdiff_t ssize_t;
+#endif //_MSC_VER
+
 typedef boost::shared_ptr<struct bladerf> bladerf_sptr;
 
 class bladerf_common
-- 
2.20.1

