From 79b53f50e42ecafa0cf2026f0274ecd1996e846c Mon Sep 17 00:00:00 2001
From: SDRplay <SDRplay@users.noreply.github.com>
Date: Tue, 14 Apr 2015 16:31:04 +0100
Subject: [PATCH 20/96] changed bufi and bufq to std::vector

---
 lib/sdrplay/sdrplay_source_c.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/sdrplay/sdrplay_source_c.h b/lib/sdrplay/sdrplay_source_c.h
index c1a3099..8eadf43 100644
--- a/lib/sdrplay/sdrplay_source_c.h
+++ b/lib/sdrplay/sdrplay_source_c.h
@@ -122,8 +122,8 @@ private:
 
    sdrplay_dev_t *_dev;
 
-   short _bufi[504];
-   short _bufq[504];
+   std::vector< short > _bufi;
+   std::vector< short > _bufq;
    int _buf_offset;
    boost::mutex _buf_mutex;
 
-- 
2.11.0

