From 3793a5e8b1b97cb78e21c807d21bca2a406f97d3 Mon Sep 17 00:00:00 2001
From: Dimitri Stolnikov <horiz0n@gmx.net>
Date: Tue, 12 May 2015 23:29:05 +0200
Subject: [PATCH 023/131] updated AUTHORS

---
 AUTHORS | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/AUTHORS b/AUTHORS
index 8bcfb94..9d07412 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -2,3 +2,5 @@ Dimitri Stolnikov <horiz0n@gmx.net>
 Steve Markgraf <steve@steve-m.de>
 Hoernchen <la@tfc-server.de>
 Nuand LLC folks
+Josh Blum
+SDRplay Ltd.
-- 
2.20.1

