From b226b85fa82f94dfc3d57d1083ea1fd6b4c62d13 Mon Sep 17 00:00:00 2001
From: Dimitri Stolnikov <horiz0n@gmx.net>
Date: Tue, 12 May 2015 23:59:32 +0200
Subject: [PATCH 026/131] README: move sdrplay line to the same position as in
 grc template

---
 README | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/README b/README
index 8c50cef..ee5019e 100644
--- a/README
+++ b/README
@@ -7,6 +7,7 @@ as well supports:
  * RTL2832U based DVB-T dongles through librtlsdr
  * RTL-TCP spectrum server (see librtlsdr project)
  * MSi2500 based DVB-T dongles through libmirisdr
+ * SDRplay RSP through SDRplay API library
  * gnuradio .cfile input through libgnuradio-blocks
  * RFSPACE SDR-IQ, SDR-IP, NetSDR (incl. X2 option)
  * AirSpy Wideband Receiver through libairspy
@@ -14,7 +15,6 @@ as well supports:
  * Nuand LLC bladeRF through libbladeRF library
  * Ettus USRP Devices through Ettus UHD library
  * Fairwaves UmTRX through Fairwaves' fork of UHD
- * SDRplay RSP through SDRplay API library
 
 By using the OsmoSDR block you can take advantage of a common software api in
 your application(s) independent of the underlying radio hardware.
-- 
2.20.1

