From 275e6aed19b9ba8563bffd318a227a1196e1da2c Mon Sep 17 00:00:00 2001
From: Dimitri Stolnikov <horiz0n@gmx.net>
Date: Wed, 13 May 2015 00:16:54 +0200
Subject: [PATCH 27/66] airspy: comment on used gain values

---
 lib/airspy/airspy_source_c.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/airspy/airspy_source_c.cc b/lib/airspy/airspy_source_c.cc
index c5a389b..4daea8d 100644
--- a/lib/airspy/airspy_source_c.cc
+++ b/lib/airspy/airspy_source_c.cc
@@ -448,6 +448,8 @@ osmosdr::gain_range_t airspy_source_c::get_gain_range( size_t chan )
 
 osmosdr::gain_range_t airspy_source_c::get_gain_range( const std::string & name, size_t chan )
 {
+  /* They don't spec any gain values in dB so we simply use gain stage indices for now. */
+
   if ( "LNA" == name ) {
     return osmosdr::gain_range_t( 0, 15, 1 );
   }
-- 
2.1.4

