From 5dca65674565b6e44bf2e790430d5bb7370f2327 Mon Sep 17 00:00:00 2001
From: Heikki Hannikainen <hessu@hes.iki.fi>
Date: Tue, 24 Feb 2015 11:37:18 +0200
Subject: [PATCH 34/48] style update to match surroundings

Signed-off-by: Heikki Hannikainen <hessu@hes.iki.fi>
---
 lib/hackrf/hackrf_source_c.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/hackrf/hackrf_source_c.cc b/lib/hackrf/hackrf_source_c.cc
index d3b3d86..d916572 100644
--- a/lib/hackrf/hackrf_source_c.cc
+++ b/lib/hackrf/hackrf_source_c.cc
@@ -146,8 +146,8 @@ hackrf_source_c::hackrf_source_c (const std::string &args)
   }
 
   _dev = NULL;
-  if (hackrf_serial)
-    ret = hackrf_open_by_serial(hackrf_serial->c_str(), &_dev);
+  if ( hackrf_serial )
+    ret = hackrf_open_by_serial( hackrf_serial->c_str(), &_dev );
   else
     ret = hackrf_open( &_dev );
   HACKRF_THROW_ON_ERROR(ret, "Failed to open HackRF device")
-- 
2.1.4

