From 69181b0e858c109d1471d4a1c9a2c87f732e8df6 Mon Sep 17 00:00:00 2001
From: Heikki Hannikainen <hessu@hes.iki.fi>
Date: Tue, 24 Feb 2015 11:37:32 +0200
Subject: [PATCH 035/131] hackrf_sink: support hackrf_open_by_serial

Signed-off-by: Heikki Hannikainen <hessu@hes.iki.fi>
---
 lib/hackrf/hackrf_sink_c.cc | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/lib/hackrf/hackrf_sink_c.cc b/lib/hackrf/hackrf_sink_c.cc
index 60ef5ce..a052093 100644
--- a/lib/hackrf/hackrf_sink_c.cc
+++ b/lib/hackrf/hackrf_sink_c.cc
@@ -168,9 +168,13 @@ hackrf_sink_c::hackrf_sink_c (const std::string &args)
     _bandwidth(0)
 {
   int ret;
+  std::string *hackrf_serial = NULL;
 
   dict_t dict = params_to_dict(args);
 
+  if (dict.count("hackrf") && dict["hackrf"].length() > 0)
+    hackrf_serial = &dict["hackrf"];
+
   _buf_num = 0;
 
   if (dict.count("buffers"))
@@ -189,7 +193,10 @@ hackrf_sink_c::hackrf_sink_c (const std::string &args)
   }
 
   _dev = NULL;
-  ret = hackrf_open( &_dev );
+  if ( hackrf_serial )
+    ret = hackrf_open_by_serial( hackrf_serial->c_str(), &_dev );
+  else
+    ret = hackrf_open( &_dev );
   HACKRF_THROW_ON_ERROR(ret, "Failed to open HackRF device")
 
   uint8_t board_id;
-- 
2.20.1

