From 254e1b1981bbe920b03c1a8e50b67509739231f3 Mon Sep 17 00:00:00 2001
From: Heikki Hannikainen <hessu@hes.iki.fi>
Date: Sat, 30 May 2015 14:30:03 +0300
Subject: [PATCH 41/48] hackrf_source: adjust error message

Signed-off-by: Heikki Hannikainen <hessu@hes.iki.fi>
---
 lib/hackrf/hackrf_source_c.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/hackrf/hackrf_source_c.cc b/lib/hackrf/hackrf_source_c.cc
index 33ab62d..64b8f14 100644
--- a/lib/hackrf/hackrf_source_c.cc
+++ b/lib/hackrf/hackrf_source_c.cc
@@ -155,7 +155,7 @@ hackrf_source_c::hackrf_source_c (const std::string &args)
           dev_index = boost::lexical_cast< unsigned int >( hackrf_serial );
         } catch ( std::exception &ex ) {
           throw std::runtime_error(
-                "Failed to use '" + hackrf_serial + "' as HackRF device index: " + ex.what());
+                "Failed to use '" + hackrf_serial + "' as HackRF device index number: " + ex.what());
         }
         
         hackrf_device_list_t *list = hackrf_device_list();
-- 
2.1.4

