From f33f30815a7b9494315d4782245447ebca21c02e Mon Sep 17 00:00:00 2001
From: Dimitri Stolnikov <horiz0n@gmx.net>
Date: Mon, 1 Jun 2015 22:24:32 +0200
Subject: [PATCH 044/131] hackrf: use #ifdef just like in hackrf_sink_c

---
 lib/hackrf/hackrf_source_c.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/hackrf/hackrf_source_c.cc b/lib/hackrf/hackrf_source_c.cc
index 13df11e..81ddad6 100644
--- a/lib/hackrf/hackrf_source_c.cc
+++ b/lib/hackrf/hackrf_source_c.cc
@@ -401,7 +401,7 @@ std::vector<std::string> hackrf_source_c::get_devices()
     _usage++;
   }
 
-#if LIBHACKRF_HAVE_DEVICE_LIST
+#ifdef LIBHACKRF_HAVE_DEVICE_LIST
   hackrf_device_list_t *list = hackrf_device_list();
   
   for (unsigned int i = 0; i < list->devicecount; i++) {
-- 
2.20.1

