From 64cfd98c5b2a8508715a1d48b09daf85a96049ee Mon Sep 17 00:00:00 2001
From: Dimitri Stolnikov <horiz0n@gmx.net>
Date: Sun, 10 Jan 2016 22:27:42 +0100
Subject: [PATCH 063/131] grc: update grc generator

---
 grc/gen_osmosdr_blocks.py | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/grc/gen_osmosdr_blocks.py b/grc/gen_osmosdr_blocks.py
index 36210f8..d29de3e 100644
--- a/grc/gen_osmosdr_blocks.py
+++ b/grc/gen_osmosdr_blocks.py
@@ -218,6 +218,9 @@ While primarily being developed for the OsmoSDR hardware, this block as well sup
  * gnuradio .cfile input through libgnuradio-blocks
  * RFSPACE SDR-IQ, SDR-IP, NetSDR (incl. X2 option)
  * AirSpy Wideband Receiver through libairspy
+#end if
+#if $sourk == 'sink':
+ * gnuradio .cfile output through libgnuradio-blocks
 #end if
  * CCCamp 2015 rad1o Badge through libhackrf
  * Great Scott Gadgets HackRF through libhackrf
@@ -252,8 +255,12 @@ Lines ending with ... mean it's possible to bind devices together by specifying
   file='/path/to/your file',rate=1e6[,freq=100e6][,repeat=true][,throttle=true] ...
   netsdr=127.0.0.1[:50000][,nchan=2]
   sdr-ip=127.0.0.1[:50000]
+  cloudiq=127.0.0.1[:50000]
   sdr-iq=/dev/ttyUSB0
-  airspy=0[,bias=0|1]
+  airspy=0[,bias=0|1][,linearity][,sensitivity]
+#end if
+#if $sourk == 'sink':
+  file='/path/to/your file',rate=1e6[,freq=100e6][,append=true][,throttle=true] ...
 #end if
   redpitaya=192.168.1.100[:1001]
   hackrf=0[,buffers=32][,bias=0|1][,bias_tx=0|1]
-- 
2.20.1

