From da2cf6a6f262c05d4082485369191c44b4b2bcf4 Mon Sep 17 00:00:00 2001
From: Dimitri Stolnikov <horiz0n@gmx.net>
Date: Sun, 10 Jan 2016 22:28:15 +0100
Subject: [PATCH 64/66] cmake: update component name & update AUTHORS file

---
 AUTHORS            | 1 +
 lib/CMakeLists.txt | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/AUTHORS b/AUTHORS
index bc7249f..b654b84 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -5,3 +5,4 @@ Nuand LLC folks
 Josh Blum
 SDRplay Ltd.
 Pavel Demin
+Marcus Müller
diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 550c24e..a99890a 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -145,7 +145,7 @@ endif(ENABLE_FCD OR ENABLE_FCDPP)
 ########################################################################
 # Setup File component
 ########################################################################
-GR_REGISTER_COMPONENT("IQ File Source" ENABLE_FILE GNURADIO_BLOCKS_FOUND)
+GR_REGISTER_COMPONENT("IQ File Source & Sink" ENABLE_FILE GNURADIO_BLOCKS_FOUND)
 if(ENABLE_FILE)
 GR_INCLUDE_SUBDIRECTORY(file)
 endif(ENABLE_FILE)
-- 
2.1.4

