From 43df1c98c4110a4d3b66ce867da7c21dbf1e6d58 Mon Sep 17 00:00:00 2001
From: Dimitri Stolnikov <horiz0n@gmx.net>
Date: Mon, 27 Feb 2017 23:03:09 +0100
Subject: [PATCH 083/131] rfspace: use boost::chrono for sleeping

---
 lib/rfspace/rfspace_source_c.cc | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/rfspace/rfspace_source_c.cc b/lib/rfspace/rfspace_source_c.cc
index 74708c6..80f34df 100644
--- a/lib/rfspace/rfspace_source_c.cc
+++ b/lib/rfspace/rfspace_source_c.cc
@@ -343,8 +343,10 @@ rfspace_source_c::rfspace_source_c (const std::string &args)
 
   }
 
+  /* Wait 10 ms before sending queries to device (required for networked radios). */
+  boost::this_thread::sleep_for(boost::chrono::milliseconds(10));
+
   /* request & print device information */
-  usleep(10000);
 
   std::vector< unsigned char > response;
 
-- 
2.20.1

