From ea6cc4beb9052d28c7fa086fa65d4f45672a1202 Mon Sep 17 00:00:00 2001
From: Ryan Tucker <ryan.tucker@nuand.com>
Date: Thu, 20 Jul 2017 15:03:47 -0400
Subject: [PATCH 115/131] bladerf_common: add agc= option to set gain mode

---
 lib/bladerf/bladerf_common.cc | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/lib/bladerf/bladerf_common.cc b/lib/bladerf/bladerf_common.cc
index b6e91d4..70fb591 100644
--- a/lib/bladerf/bladerf_common.cc
+++ b/lib/bladerf/bladerf_common.cc
@@ -487,6 +487,16 @@ void bladerf_common::init(dict_t &dict, bladerf_direction direction)
               << std::endl;
   }
 
+  /* Specify initial gain mode */
+  if (direction == BLADERF_RX) {
+    if (dict.count("agc")) {
+      set_gain_mode(boost::lexical_cast<bool>(dict["agc"]));
+      std::cerr << _pfx
+                << "Gain mode set to " << get_gain_mode()
+                << std::endl;
+    }
+  }
+
   /* Initialize buffer and sample configuration */
   _num_buffers = 0;
   if (dict.count("buffers")) {
-- 
2.20.1

