From 68ba383fd5f9b35d47dbc398cc4202a90a1a4c3c Mon Sep 17 00:00:00 2001
From: Ryan Tucker <ryan.tucker@nuand.com>
Date: Thu, 20 Jul 2017 16:12:49 -0400
Subject: [PATCH 116/131] bladerf: add firmware loopback mode

---
 lib/bladerf/bladerf_common.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/bladerf/bladerf_common.cc b/lib/bladerf/bladerf_common.cc
index 70fb591..d099c29 100644
--- a/lib/bladerf/bladerf_common.cc
+++ b/lib/bladerf/bladerf_common.cc
@@ -187,6 +187,8 @@ void bladerf_common::set_loopback_mode(const std::string &loopback)
     mode = BLADERF_LB_RF_LNA2;
   } else if (loopback == "rf_lna3") {
     mode = BLADERF_LB_RF_LNA3;
+  } else if (loopback == "firmware") {
+    mode = BLADERF_LB_FIRMWARE;
   } else if (loopback == "ad9361_bist") {
     mode = BLADERF_LB_AD9361_BIST;
   } else if (loopback == "none") {
-- 
2.20.1

